src/sysupdate/sysupdate-transfer.c

```diff
@@ -784,30 +791,236 @@ static void compile_pattern_fields(
         memcpy(ret->sha256sum, i->metadata.sha256sum, sizeof(ret->sha256sum));
 }
 
+typedef struct CalloutContext {
+        const Transfer *transfer;
+        const Instance *instance;
+        TransferProgress callback;
+        PidRef pid;
+        const char *name;
+        void* userdata;
+} CalloutContext;
+
+
+static CalloutContext *callout_context_free(CalloutContext *ctx)
+{
```

Comment:
coding style, we place the opening `{` on the same line as the function name, see:

https://systemd.io/CODING_STYLE/#formatting

Reply:
Whoops, sorry, force of habit!